src/views/tutorial-view/components/image-config/index.tsx

```diff
@@ -0,0 +1,72 @@
+import { ReactElement, ReactNode } from 'react'
+import getValidatedImgChild from './get-validated-image-child'
+import MdxImg from '../mdx-img'
+// import classNames from 'classnames'
+import s from './image-config.module.css'
+
+/**
+ * Configure extra settings for images via MDX.
+ * Designed to wrap a single markdown image.
+ */
+export default function ImageConfig({
+  /** Hide borders (temporarily configurable while we remove baked-in borders from existing images) */
+  hideBorder = false,
+  /** A caption to display below the image. */
+  caption,
+  /** An MDX `img`, e.g. `![some alt text](path/to/img.jpg)` */
+  children,
+}: {
+  caption?: string
+  hideBorder?: boolean
+  children: ReactNode
+}): ReactElement {
+  //  emit a warning if no meaningful props are present
+  if (!hideBorder && !caption) {
+    console.warn(
+      `Warning: <ImageConfig /> was initialized with default props. Please remove <ImageConfig /> if configuration is not needed.`
+    )
+  }
+
+  // TODO: fully deprecate the `hideBorder` prop, then remove this warning
+  // Task: https://app.asana.com/0/0/1200167326263350/f
+  if (hideBorder) {
+    console.warn(
+      `Warning: <ImageConfig /> was initialized with a deprecated prop "hideBorder". This property will be deprecated in the near future. If you've provided an image caption, a border will be displayed even if "hideBorder" is set. If your image has a border built into it, please crop it to remove it.`
+    )
+  }
+
+  const imgElement = getValidatedImgChild(children)
+  // Validate that the { src, alt, title } props are present
+  const { src, alt, title } = imgElement.props
+  // Validate that src is present, and is a non-empty string
+  if (typeof src !== 'string' || src == '') {
+    throw new Error(
+      `Images in MDX must have an source defined. Use the format ![some alt text](path/to/img.jpg). Image details: ${JSON.stringify(
+        { src, alt, title }
+      )} `
+    )
+  }
```

Comment:
What do you think about adding a console warning for `alt` being `undefined` or `null`? Even if an image is decorative (doesn't require text to be written out), it's important for `alt` to be set to an empty string in those cases. Reference: https://www.w3.org/WAI/tutorials/images/decorative/

Reply:
💖 Great callout - I've added this in!

Unfortunately led to a bit of a rabbit hole - the one case that looks difficult is for those cases where we want to set `alt=""`. In JSX this would be trivial. In MDX though, it looks like `![](/some/presentational.jpg)` gives `alt` as undefined, and I'm not sure how to get `alt` to be an empty string (other than using `<img>` directly). Have added [a task to discuss](https://app.asana.com/0/1201987349274776/1202042900781423/f), I feel like it's something we could address in more detail outside the scope of this PR.

Let me know what you think! :🙇‍♂️